compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -1286,8 +1286,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     field.vis.is_accessible_from(tcx.parent_module(pat.hir_id).to_def_id(), tcx)
                 })
                 .collect();
-            if non_exhaustive {
-                self.non_exhaustive_reachable_pattern(pat, &accessible_unmentioned_fields, adt_ty)
+            if non_exhaustive && !accessible_unmentioned_fields.is_empty() {
```

Comment:
I think that would still cause errors. If I follow correctly in the following example we'd lint for `non_exhaustive_omitted_patterns` and then fail to emit the important "some fields are missing" error because there's an `else if`.
```rust
#[non_exhaustive]
struct Foo {
	a: bool,
	b: bool
}
fn do_something(foo: Foo) {
	let Foo { a } = foo;
}
```
So we probably want `if etc && non_exhaustive && !accessible_unmentioned_fields.is_empty()`

Reply:
Ah wait never mind, the lack of `, ..` would be caught earlier. Still, could you swap the two conditions, so the `!etc` test is done first? That'd be clearer in case things change later.

Reply:
Ahh yeah, good point, I should probably also add tests then since it is more complicated than originally seemed.

Reply:
How do you feel about calling `etc` -> `struct_wild` or `struct_rest` or `rest_pat` or `rest_wild`? `etc` is almost too short to be easily spotted :shrug: I didn't do it yet, I'm still kinda iffy about any of those being a better name.

Reply:
oh yea please do, took me a while to figure out what it was too. The rust book doesn't give a name to that syntax sadly. Maybe `has_etc_pat` or `has_rest_pat`?